File: p7.c
#include <stdio.h>
#include <stdlib.h>

/* TODO: check for overflow. */

int32_t reverse(int32_t);

int main()
{
	int32_t foo = 1534236469;
	printf("%d\n", reverse(foo));
	return 0;
}

int32_t reverse(int32_t x)
{
	int32_t i, result = 0;

	for (i = 0; x; i++) {
		printf("%d, ", x);
		int d = x % 10;
		result = 10*result + d;
		x = (x - d) / 10;
	}
	return result;
}

File: python3/p7.py
class Solution:
    def reverse(self, x: int) -> int:
        if -2**31 <= x < 2**31:
            print(0)
            return int(str(x)[::-1]) if x >= 0 else -int(str(x)[1::][::-1])
        else:
            return 0

S = Solution()
print(S.reverse(1534236469))
